include/circt/Dialect/RTL/Combinatorial.td

```diff
@@ -73,6 +73,25 @@ def AndOp : UTBinRTLOp<"and", [Commutative]>;
 def OrOp  : UTBinRTLOp<"or",  [Commutative]>;
 def XorOp : UTBinRTLOp<"xor", [Commutative]>;
 
+//===----------------------------------------------------------------------===//
+// Unary Operations
+//===----------------------------------------------------------------------===//
+
+// Base class for unary primitive operators.
+class UnaryRTLOp<string mnemonic, list<OpTrait> traits = []> :
+      RTLOp<mnemonic, !listconcat(traits, [NoSideEffect])> {
+  let arguments = (ins AnySignlessInteger:$input);
+  let results = (outs AnySignlessInteger:$result);
```

Comment:
Random suggestion to improve this: MLIR allows us to specify that these always return i1, which can be used to simplify the assembly syntax.  I'll take a crack at this if I have time.